lib/Filters/GitDiffFilter.cpp

```diff
@@ -0,0 +1,97 @@
+#include "mull/Filters/GitDiffFilter.h"
+
+#include "mull/Diagnostics/Diagnostics.h"
+#include "mull/Toolchain/Runner.h"
+
+#include "llvm/IR/InstIterator.h"
+#include <llvm/IR/DebugInfoMetadata.h>
+#include <llvm/IR/DebugLoc.h>
+#include <llvm/IR/Function.h>
+#include <llvm/Support/FileSystem.h>
+
+#include <iostream>
+#include <regex>
+#include <sstream>
+#include <unistd.h>
+
+using namespace mull;
+
+static std::string getCwd() {
+  char cwd_buffer[8096];
+  if (getcwd(cwd_buffer, sizeof(cwd_buffer)) == NULL) {
+    return "";
+  }
+  return std::string(cwd_buffer);
+}
+
+GitDiffFilter::GitDiffFilter(Diagnostics &diagnostics)
+    : diagnostics(diagnostics), gitDiffInfoStorage() {}
+
+std::string GitDiffFilter::name() {
+  return "Git Diff";
+}
+
+void GitDiffFilter::activateAgainstGitBranch(const std::string &gitDiffBranch) {
+  /// Assuming that Mull is run from the project's directory.
+  const std::string gitRootPath = getCwd();
```

Comment:
I am not sure about this one: it is either communicating this in the docs that Mull has to run from the project's Git root or provide an extra argument that specifies the path to Git root.

Reply:
Let's just have it in the docs for the beginning